logos/src/logos/capacity/calibration_orchestrator.py

```
@@ -0,0 +1,353 @@
+                return provider_id, model_name
+
+        return None
+
+    def _provider_has_active_requests(self, provider_id: int) -> bool:
+        """Return True if the provider has any active inference requests."""
+        try:
+            capacity = self._facade.get_capacity_info(provider_id)
+            # OllamaCapacity.active_requests counts requests currently running
+            active: int = getattr(capacity, "active_requests", 0) or 0
+            return active > 0
+        except Exception:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)